grc/gui_qt/components/canvas/connection.py

```diff
@@ -159,5 +159,6 @@ def paint(self, painter, option, widget):
         painter.drawPath(self._arrowhead)
 
     def mouseDoubleClickEvent(self, e):
-        self.parent.connections.remove(self)
+        connection = self.core
+        connection.source_port.parent_flowgraph.connections.remove(connection)
```

Comment:
Isn't there a `connection.parent`? Or did that not work?

Reply:
you're right. I think using `connection.parent` is better than `connection.source_port.parent_flowgraph`